packages/core/admin/server/services/project-settings.js

```diff
@@ -75,7 +75,11 @@ const getProjectSettings = async () => {
 
 const uploadFiles = async (files = {}) => {
   // Call the provider upload function for each file
-  return Promise.all(Object.values(files).map(strapi.plugin('upload').provider.uploadStream));
+  return Promise.all(
+    Object.values(files)
+      .filter(file => file.stream instanceof fs.ReadStream)
+      .map(strapi.plugin('upload').provider.uploadStream)
```

Comment:
Some providers take multiple parameters (eg: [aws s3](https://github.com/strapi/strapi/blob/master/packages/providers/upload-aws-s3/lib/index.js#L45)). By mapping directly with the `uploadStream` method, you're potentially sending the file index in the collection as `customParams` .

Maybe do smth like that instead:
```js
        .map(file => strapi.plugin('upload').provider.uploadStream(file))
```

Reply:
Ouch good catch!